src/state.rs

```diff
@@ -526,127 +503,37 @@ impl DemandProxyState {
-        if resp.is_err() {
-            warn!(
-                "system dns async resolution: error response for workload {} is: {:?}",
-                &workload_uid, resp
-            );
-            return;
-        } else {
-            trace!(
-                "system dns async resolution: response for workload {} is: {:?}",
-                &workload_uid,
-                resp
-            );
-        }
-        let resp = resp.unwrap();
-        let mut dns_refresh_rate = std::time::Duration::from_secs(u64::MAX);
-        let ips = HashSet::from_iter(resp.as_lookup().record_iter().filter_map(|record| {
-            if record.record_type().is_ip_addr() {
-                let record_ttl = u64::from(record.ttl());
-                if let Some(ipv4) = record.data().unwrap().as_a() {
-                    if record_ttl < dns_refresh_rate.as_secs() {
-                        dns_refresh_rate = std::time::Duration::from_secs(record_ttl);
-                    }
-                    return Some(IpAddr::V4(ipv4.0));
-                }
-                if let Some(ipv6) = record.data().unwrap().as_aaaa() {
-                    if record_ttl < dns_refresh_rate.as_secs() {
-                        dns_refresh_rate = std::time::Duration::from_secs(record_ttl);
-                    }
-                    return Some(IpAddr::V6(ipv6.0));
-                }
-                return None;
+        let resp = match self.dns_resolver.lookup_ip(hostname.as_str()).await {
+            Err(err) => {
+                warn!(?err,%hostname,"dns lookup failed");
+                return Err(Error::NoResolvedAddresses(workload_uid.to_string()));
             }
-            None
-        }));
-        if ips.is_empty() {
-            // if we have no DNS records with a TTL to lean on; lets try to refresh again in 60s
-            dns_refresh_rate = std::time::Duration::from_secs(60);
-        }
-        let now = std::time::Instant::now();
-        let rdns = ResolvedDns {
-            hostname: hostname.to_owned(),
-            ips,
-            initial_query: Some(now),
-            dns_refresh_rate,
+            Ok(resp) => resp,
         };
-        state.set_ips_for_hostname(hostname, rdns);
-    }
-
-    pub fn set_ips_for_hostname(&self, hostname: Strng, rdns: ResolvedDns) {
-        self.state
-            .write()
-            .unwrap()
-            .resolved_dns
-            .by_hostname
-            .insert(hostname, rdns);
-    }
-
-    pub fn get_ips_for_hostname(&self, hostname: &Strng) -> Option<ResolvedDns> {
-        self.state
-            .read()
-            .unwrap()
-            .resolved_dns
-            .by_hostname
-            .get(hostname)
-            .filter(|rdns| {
-                rdns.initial_query.is_some()
-                    && rdns.initial_query.unwrap().elapsed() < rdns.dns_refresh_rate
-            })
-            .cloned()
+        warn!(%hostname,"dns lookup complete {resp:?}");
```

Comment:
warn seems a bit high for this, probably info or maybe even lower is more appropriate

Reply:
thanks, probably was a debug log while I was testing